sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -0,0 +1,175 @@
+ * @internal
+ */
+export const powerShellPublicErrorMessages = {
+  login:
+    "Please run 'Connect-AzAccount' from powershell to authenticate before using this credential.",
+  installed: `Az.Account module >= 2.2.0 is not installed. Install the Azure Az PowerShell module with: "Install - Module - Name Az - Scope CurrentUser - Repository PSGallery - Force".`
+};
+
+// PowerShell Azure User not logged in error check.
+const isLoginError = (err: Error) => err.message.match(`(.*)${powerShellErrors.login}(.*)`);
+
+// Az Module not Installed in Azure PowerShell check.
+const isNotInstallError = (err: Error) => err.message.match(powerShellErrors.installed);
+
+/**
+ * Optional parameters for the {@link AzurePowerShellCredential} class.
+ */
+export interface AzurePowerShellCredentialOptions {
+  /**
+   * If specified, this credential will use the legacy `powershell` command instead of the newest `pwsh`.
+   */
+  useLegacyPowerShell?: boolean;
+}
+
+/**
+ * This credential will use the currently logged-in user login information via the Azure Power Shell command line tool.
+ * To do so, it will read the user access token and expire time with Azure Power Shell command `Get-AzAccessToken -ResourceUrl {ResourceScope}`.
+ * To be able to use this credential, ensure that:
+ * - Install the Azure Az PowerShell module with: `Install-Module -Name Az -Scope CurrentUser -Repository PSGallery -Force`.
+ * - You have already logged in via the 'az' tool using the command "Connect-AzAccount" from the command line.
+ */
+export class AzurePowerShellCredential implements TokenCredential {
+  private useLegacyPowerShell: boolean;
+
+  /**
+   * Creates an instance of the ClientCertificateCredential with the details
+   * needed to authenticate against Azure Active Directory with a certificate.
+   *
+   * @param useLegacyPowerShell - The flag indicating if legacy powershell should be used for authentication.
+   */
+  constructor(options?: AzurePowerShellCredentialOptions) {
+    this.useLegacyPowerShell = Boolean(options?.useLegacyPowerShell);
+  }
+  /**
+   * Gets the access token from Azure Power Shell
+   * @param resource - The resource to use when getting the token
+   */
+  protected async getAzurePowerShellAccessToken(
+    resource: string
+  ): Promise<{ Token: string; ExpiresOn: string }> {
+    let pwshCommand: string;
+
+    if (this.useLegacyPowerShell) {
+      pwshCommand = formatCommand("powershell");
+    } else {
+      pwshCommand = formatCommand("pwsh");
+    }
+
+    try {
+      await processUtils.exists(pwshCommand);
+    } catch (e) {
+      throw new Error(
+        `Unable to execute "${pwshCommand}". Ensure that it is installed in your system.`
+      );
+    }
+
+    const output = runCommands([
+      `${pwshCommand} -Command Import-Module Az.Accounts -MinimumVersion 2.2.0 -PassThru`,
+      `${pwshCommand} -Command 'Get-AzAccessToken -ResourceUrl "${resource}" | ConvertTo-Json'`
+    ]);
+
+    return JSON.parse(output[1]);
```

Comment:
If the command somehow fails and doesn't produce JSON, this will throw a nasty error message about being unable to parse. It'd probably be best to wrap this in a try/catch and throw a nicer error message.